### DISCUSSION TITLE
This is what I don't like external CLIs for

### DISCUSSION THREAD

**Discussion by xahon** [tauri-apps/tauri#9366] [General]
I'm not sure it's an expected behaviour to silently rewrite manifest, guess it should panic on the spot and telling that you need to define features in config (I guess) first

UPD: I mean this line https://github.com/tauri-apps/tauri/blob/36b4c12497fbe636066f4848c6877b3ab6cc892e/tooling/cli/src/interface/rust.rs#L123

**Comment by FabianLars** [comment]
While the cli has no flag for it _yet_, it will be automatically disabled if you use [workspace inheritance](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table) for tauri and tauri-build.
If you use `cargo run` instead of the tauri cli (will break some features like `beforeDevCommand` and `resources`) it will panic instead too.

**Reply by xahon** [reply]
Still seems to be too dangerous from UX-side standpoint for a regular user like me. I used `cargo tauri dev` and got into the situation it does some rewrites in my project manifest file

**Reply by FabianLars** [reply]
"dangerous" sounds a bit harsh to me. As far as we are aware it did not actually break any project so far. But it really shows that this was created for non-rust users primarily.

With the removal of the allowlist we could also just remove it completely and ask users to add the feature flags they need themselves (iirc only 4 are left). cc @lucasfernog @amrbashir (i have a feeling they'll disagree with me so i'm gonna ping them early so that i don't waste my time thinking about it 😂 )
